### PR TITLE
style: sever some dependencies between primitives and vm-runner

### PR DESCRIPTION
**Pull request by nagisa**
Most notably things like conversions between error types and re-exports of e.g. TrieNodesCount now no longer introduce a dependency edge.

There is still a dependency on a few instances however, but those are much less clear cut on how to resolve. In particular at least `ProfileDataV3` is in an icky place. It should be split up into a serialized schema (just like the errors were) and the simple structure that `near-vm-runner` produces with a conversion in some integration crate downstream…

### CODE REVIEW

**Review by frol — APPROVED** [review]
Great initiative. I’m hopeful that with such incremental steps we can make nearcore project more maintainable and crates of it reusable in the whole ecosystem, and as such boost speed of innovation

File: core/store/src/trie/mod.rs [hunk 1]
```diff
@@ -1493,6 +1492,25 @@ impl TrieAccess for Trie {
     }
 }
 
+/// Counts trie nodes reads during tx/receipt execution for proper storage costs charging.
+#[derive(Debug, PartialEq)]
+pub struct TrieNodesCount {
```
Comment (Longarithm) [inline core/store/src/trie/mod.rs:1497]: Why not to import it from near_vm_runner temporarily, if we depend on it anyway?

File: runtime/runtime/src/actions.rs [hunk 2]
```diff
@@ -271,7 +271,9 @@ pub(crate) fn action_function_call(
         }
         // Update action result with the abort error converted to the
         // transaction runtime's format of errors.
-        let action_err: ActionError = ActionErrorKind::FunctionCallError(err.into()).into();
+        let action_err: ActionError =
+            ActionErrorKind::FunctionCallError(crate::conversions::function_call_error::cvt(err))
```
Comment (Longarithm) [inline runtime/runtime/src/actions.rs:275]: Is method `cvt` a standard as well, can we continue using `into` here which is more readable?

**Review by Longarithm — APPROVED** [review]
Approving to unblock but leaving two questions.

File: core/store/src/trie/mod.rs (hunk 1 above)
Comment (nagisa) [reply core/store/src/trie/mod.rs:1497]: So the most sensible option is to really have two types. One is an "output" of `near-store` and the other is used (but doesn’t need to be!) in the interface that `near-vm-runner` exposes via `trait Externals`.

**Pull request merged**